addons/web/static/src/model/relational_model/dynamic_list.js

```diff
@@ -200,6 +204,37 @@ export class DynamicList extends DataPoint {
         return this.model.mutex.exec(() => this._toggleArchive(isSelected, false));
     }
 
+    toggleArchiveWithConfirmation(isArchived, dialogProps = {}) {
```

Comment:
Suggestion:

`toggleArchiveWithConfirmation(archive, dialogProps = {}) {`

because the records aren't archived *yet*